### ISSUE TITLE
Form elements have no labels

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#742]
## Bug Description

Most of the form elements (e.g. icon buttons, text inputs, etc.) have no label (visible or `aria-label`).

Related effort: #698.

## Screenshots

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by jauyong** [comment]
noted in sprint 26 sheet

**Comment by dvoytenko** [comment]
I recommend we make `label`/`aria-label`/etc required prop for now and clean up warnings in one go.

**Comment by miina** [comment]
Additionally, the existing label-like texts are not "behaving" like labels, for example:
<img width="319" alt="Screenshot 2020-03-26 at 11 24 49" src="https://user-images.githubusercontent.com/3294597/77636644-b69e8700-6f54-11ea-9ea1-13f4b79e41cc.png">

and 

<img width="321" alt="Screenshot 2020-03-26 at 11 27 18" src="https://user-images.githubusercontent.com/3294597/77636690-c61dd000-6f54-11ea-87b6-40b2f8501a30.png">
